src/view/graph.py

```diff
@@ -175,53 +149,22 @@ def mouseReleaseEvent(self, event=None):
         if not self.graph.holdCtrl:
             self.handle.setSelected(False)
 
-    def addJunction(self, type, size, color):
+    def addJunction(self):
         """
         create a junction to fix link
 
-        Parameters
-        ----------
-        type: {'out', 'in'}
-            type of junction, if 'out' the junction will be hidden
-            the type influence the relative position of the junction in the node
-
         Return
         ------
         junction: Junction
         """
-        junction = Junction(self, type, size, color)
+        junction = Junction(self)
         self.positionChanged.connect(junction.updateLinkPos)
-        self.sizeChanged.connect(lambda: junction.setPos(*self.getJunctionRelativePosition(type)))
+        self.sizeChanged.connect(lambda: junction.setPos(self.width()/2, self.height()/2))
```

Comment:
To be clearer you can use @property decorator to use a specific name for that operation like:

    @property
    def mid_pos(self):
        return self.width()/2, self.height()/2

and then call it here something like that:

    self.sizeChanged.connect(lambda: junction.setPos(*self.mid_pos))

If you prefer having the operation here it is ok also ;)

Reply:
ok I never used this decorator!

Reply:
It can be useful to have code easier to understand 😉